tensorrt_llm/_torch/pyexecutor/kv_cache_manager_v2.py

```
@@ -2531,6 +2531,18 @@ def _prepare_draft_resources(self, scheduled_batch: ScheduledRequests):
                         f"{req.py_request_id}: could not resize to {new_cap} tokens"
                     )
 
+    def _reuse_token_source(self, req: LlmRequest) -> Sequence[int]:
+        """Beam-0 tokens for block reuse, in the form the active backend consumes.
+
+        The C++ backend ingests a zero-copy int32 view (get_tokens_view) — no per-token
+        PyLong allocation. The pure-Python backend cannot consume a numpy array, so it gets
+        a plain list (get_tokens), preserving pre-optimization behavior. ``_cpp_introspection``
+        is None exactly when the Python backend is active.
+        """
+        if _cpp_introspection is not None:
```

Ruff (0.16.1)

[error] Undefined name `_cpp_introspection`

(F821)